.github/actions/setup-go-tip/action.yml

```diff
@@ -8,9 +8,25 @@ runs:
       shell: bash
       run: |
         set -euo pipefail
-        tip=$(git ls-remote https://github.com/golang/go.git HEAD | awk '{print $1;}')
-        echo "Go Tip version: ${tip}"
-        curl -fsSL https://storage.googleapis.com/go-build-snap/go/linux-amd64/${tip}.tar.gz -o gotip.tar.gz
+        retries=10
+        wait_time=30
+        success=false
+        for ((i=1; i<=retries; i++)); do
+          tip=$(git ls-remote https://github.com/golang/go.git HEAD | awk '{print $1;}')
+          echo "Go Tip version: ${tip}"
```

Comment:
L15-16 should be outside of the loop, I'd put them right after L10

Reply:
addressed!